crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -74,6 +74,14 @@ pub enum SenderData {
         /// If false, we had simply accepted the key as this user's latest
         /// key.
         master_key_verified: bool,
+
+        #[serde(default)]
+        /// Whether the user's master key was previously verified, but the
+        /// current master key is not.
+        ///
+        /// TODO: should this be merged with the master_key_verified field and
+        /// turned into an enum?
```

Comment:
Short answer: yes, this really feels like it ought to be a tri-state.

(https://github.com/matrix-org/matrix-rust-sdk/pull/3846 added an `OwnUserIdentityVerifiedState` enum for a similar reason; not sure we should reuse that, but I think it's a pattern we should follow)

A downside is that it breaks backwards-compatibility for the crypto store :/